Changed dilation method and removed CustomTargetTableModel since it is too specific

che85 opened the pull request:
fixes #21

che85 committed:
ENH: Moved CustomTargetTableModel back to SliceTracker since this is a more specific model

che85 committed:
BUG: adding parameters for specifying dilateValue and erodeValue (issue #21):
- also changed kernel size

fedorov commented on SlicerProstate/SlicerProstateUtils/mixins.py:
```diff
@@ -246,13 +246,13 @@ def get3DEuclideanDistance(pos1, pos2):
-    dilateErode.SetKernelSize(12, 12, 1)
+    dilateErode.SetDilateValue(dilateValue)
+    dilateErode.SetErodeValue(erodeValue)
+    dilateErode.SetKernelSize(3, 3, 3)
```
Note that kernel is specified in voxels, and prostate imaging is very anisotropic (0.5x0.5x4). The original setting, in effect, after multiplying by the voxel spacing, results in a more regular kernel shape.

che85 replied:
So how would it look like then?

fedorov replied:
I would think kernelSize should be as it is right now, and be configurable via a parameter too.

If I understand the operation correctly, kernel in mm was 12_.5 x 12_.5 x 1*4 = 6x6x4, and with your proposed change it will be 1.5x1.5x12 mm!

che85 replied:
The first Version of SliceTracker had the following:

https://github.com/SlicerProstate/SliceTracker/blob/1e48b64d8c0c334c1760ea1e9ac5fc3d700db268/SliceTracker.py#L2581-L2584

lassoan replied:
FYI, the segment editor module computes the kernel size from the pixel spacing:

https://github.com/Slicer/Slicer/blob/master/Modules/Loadable/Segmentations/EditorEffects/Python/SegmentEditorMarginEffect.py#L61..L70

The kernel size must be an odd number, otherwise the output image may get shifted by one voxel.

che85 replied:
@lassoan I didn't know about that.

fedorov commented on SlicerProstate/SlicerProstateUtils/mixins.py:
```diff
@@ -246,13 +246,13 @@ def get3DEuclideanDistance(pos1, pos2):
 
   @staticmethod
-  def dilateMask(label):
+  def dilateMask(label, dilateValue, erodeValue=0):
```
should dilateValue have a default too?

che85 replied:
What would you expect as the default value here?

fedorov replied:
1.0?

che85 committed:
ENH: Added default value for dilateValue (issue #21)

che85 committed:
ENH: Adding margin to dilation instead of using kernel size directly (issue #21)
- as proposed by @lassoan computing the kernel size from the pixel spacing and margin [mm]
- margin is set to 5mm by default

che85 committed:
ENH: Changed implementation of SmartDICOMReceiver which should not directly access slicer statusbar
- instead only invocation of an status changed event
- added new widget class CustomStatusProgressbar which can be used to show status and progressbar in bottom status bar of slicer
- also improved IncomingDataMessageBox by placing checkbox on a better position
related to https://github.com/SlicerProstate/SliceTracker/issues/176
